components/screens/ledger-connect/LedgerConnectForSign.tsx

```diff
@@ -1,41 +1,25 @@
 import useLedgerTransport from "@/hooks/useLedgerTransport";
-import { useNavigate, useSearchParams } from "react-router-dom";
-import { useEffect, useRef } from "react";
+import { useEffect } from "react";
 import Header from "@/components/GeneralHeader";
 import ledgerConnectingImage from "@/assets/images/ledger-connecting.png";
 import { twMerge } from "tailwind-merge";
+import LedgerConnectForSignFailed from "@/components/screens/ledger-connect/LedgerConnectForSignFailed";
 
 export default function LedgerConnectForSign({
   showPrevious = true,
   showClose = true,
+  onBack,
+  onClose,
 }: {
   showPrevious?: boolean;
   showClose?: boolean;
+  onBack?: () => void;
+  onClose?: () => void;
 }) {
   const { transport, connect, isAppOpen, isConnecting } = useLedgerTransport();
-  const [searchParams] = useSearchParams();
-  const navigate = useNavigate();
-  const redirect = searchParams.get("redirect");
-  const state = searchParams.get("state");
   const [showConnect, setShowConnect] = useState(true);
   const [isPreviousConnected, setIsPreviousConnected] = useState(false);
-
-  const redirectBack = useCallback(() => {
-    if (!redirect) {
-      return;
-    }
-
-    const [pathname, search] = redirect.split("?");
-    navigate(
-      {
-        pathname: pathname,
-        search: search,
-      },
-      {
-        state: state ? JSON.parse(decodeURIComponent(state)) : undefined,
-      },
-    );
-  }, [redirect, state]);
```

Comment:
I decided not to use redirect process since it is expensive to handle multiple states on pages, especially on current form step design.